src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs

```diff
@@ -25,13 +25,14 @@ internal class CommandSwitch
 
         internal CommandSwitch(string name, string abbreviation, SwitchType switchType)
         {
-            //ensure that either name doesn't start with '/' or '-'
+            //ensure that either name doesn't start with '--' or '-'
             //also convert to lower-case
-            if ((name[0] == '/') || (name[0] == '-'))
-                _name = (name.Substring(1)).ToLower(CultureInfo.InvariantCulture);
+            if (name.Length > 1 && name[0] == '-' && name[1] == '-')
```

Comment:
This code can be made a lot more readable. It's not in a hot code path so no need to use individual character comparisons.
```c#
if (name.StartsWith("--"))
    name = name.Substring(2);

_name = name.ToLower(CultureInfo.InvariantCulture);

if (abbreviaton.StartsWith('-'))
    abbreviation = abbreviation.Substring(1);

_abbreviation = abbreviation.ToLower(CultureInfo.InvariantCulture);